### ISSUE TITLE
upload key is empty

### ISSUE THREAD

**Issue by ec1oud** [hansendc/eyefi-config#7]
```
$ sudo ./eyefi-config -u
located EyeFi card at: '/mnt/usb'
card upload key (len: 0): ''
```

What does it mean?  My card was activated, I just haven't used it for years.

**Comment by winkler-winsen** [comment]
What kind of card did you use?

**Comment by elperepat** [comment]
Hi!

I know I'm late to the party, but I just bought an X2-Pro 8GB on ebay and have the same problem. 

I already have 1 working X2-Pro 8GB which I can get debug info from and I can compare the response from the other if needed.

The non-working one is version '5.0007 Feb 25 2012 11:08:35'
The working one is '5.2010 Aug 27 2013 18:12:44'

I get a MAC from both, can configure wireless too. In fact the upload key is the only thing not working.

Any idea what I could do to get the upload key?

**Comment by RamSet** [comment]
You cannot get the key with this (or any for that matter) software. That
one will not be usable as an Eye-Fi card.

**Comment by elperepat** [comment]
With my first card, I can use the -u argument and get my upload key:

> ~/eyefi-config-master/eyefi-config -u
> located EyeFi card at: '/media/elperepat/Eye-Fi'
> card upload key (len: 32): 'e0c54747431619354cd06489a9a8e906'
[output from eyefi-config -u -d 5](https://github.com/hansendc/eyefi-config/files/5720305/good.txt)

The one I just bought gives:

> ~/eyefi-config-master/eyefi-config -u
> located EyeFi card at: '/media/elperepat/Eye-Fi'
> card upload key (len: 0): ''
[output from eyefi-config -u -d 5](https://github.com/hansendc/eyefi-config/files/5720304/bad.txt)

That key is what I need insert in the settings.xml file for the eyefi2utility.

You mean this card has never been activated and therefore the upload key was never generated?

**Comment by RamSet** [comment]
Yep. That means that the upload key cannot be read/decoded. It does not
work with certain (later) firmwares.

It will always show 0.

This code has not been maintained for years now so any hope of retrieving
that “newer firmware” card’s upload key is close to nil.

**Comment by elperepat** [comment]
The one I bought on ebay has an earlier firmware than my old one (which I probably upgraded the firmware before eye-fi went belly up).

I guess I just bought an expensive 8gb card.

Thanks!
